cmd/influx_inspect/verify_seriesfile/verify_seriesfile.go

```diff
@@ -0,0 +1,298 @@
+// Package verify_seriesfile verifies integrity of series files.
+package verify_seriesfile
+
+import (
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sort"
+
+	"github.com/influxdata/influxdb/logger"
+	"github.com/influxdata/influxdb/tsdb"
+	"go.uber.org/zap"
+)
+
+// Command represents the program execution for "influx_inspect verify-seriesfile".
+type Command struct {
+	Stderr io.Writer
+	Stdout io.Writer
+}
+
+// NewCommand returns a new instance of Command.
+func NewCommand() *Command {
+	return &Command{
+		Stderr: os.Stderr,
+		Stdout: os.Stdout,
+	}
+}
+
+// Run executes the command.
+func (cmd *Command) Run(args ...string) error {
+	fs := flag.NewFlagSet("verify-seriesfile", flag.ExitOnError)
+	fs.SetOutput(cmd.Stdout)
+
+	dataDir := fs.String("dir", os.Getenv("HOME")+"/.influxdb", "data directory.")
+	dbPath := fs.String("db", "", "database path. overrides data directory.")
+	filePath := fs.String("file", "", "series file path. overrides db and data directory.")
+
+	if err := fs.Parse(args); err != nil {
+		return err
+	}
+
+	return cmd.run(*dataDir, *dbPath, *filePath)
+}
+
+// run performs all of the validations, preferring a filePath, then a dbPath,
+// and then a dataDir.
+func (cmd *Command) run(dataDir, dbPath, filePath string) error {
```

Comment:
Could we change this up a bit so there is a programatic entrypoint to the tool? I would like eventually for all tools to be able to be run from inside an InfluxDB (or other) process. I don't think that's currently possible if the exported method handles flag stuff. Feels like this method should be exported (and I guess renamed not to clash with the one that handles flags?)

Reply:
I'll do some refactoring to make this more consumable externally.